akka-cluster-sharding-typed/src/test/java/jdocs/delivery/ShardingDocExample.java

```diff
@@ -0,0 +1,385 @@
+
+            return Behaviors.receive(Command.class)
+                .onMessage(WrappedRequestNext.class, w -> new Active(context, w.next))
+                .onMessage(
+                    UpdateTodo.class,
+                    command -> {
+                      // not hooked up with shardingProducerController yet
+                      command.replyTo.tell(Response.REJECTED);
+                      return Behaviors.same();
+                    })
+                .build();
+          });
+    }
+
+    static class Active extends AbstractBehavior<TodoService.Command> {
+
+      private ShardingProducerController.RequestNext<TodoList.Command> requestNext;
+
+      Active(
+          ActorContext<Command> context,
+          ShardingProducerController.RequestNext<TodoList.Command> requestNext) {
+        super(context);
+        this.requestNext = requestNext;
+      }
+
+      @Override
+      public Receive<Command> createReceive() {
+        return newReceiveBuilder()
+            .onMessage(WrappedRequestNext.class, this::onRequestNext)
+            .onMessage(UpdateTodo.class, this::onUpdateTodo)
+            .onMessage(Confirmed.class, this::onConfirmed)
+            .onMessage(TimedOut.class, this::onTimedOut)
+            .build();
+      }
+
+      private Behavior<Command> onRequestNext(WrappedRequestNext w) {
+        requestNext = w.next;
+        return this;
+      }
+
+      private Behavior<Command> onUpdateTodo(UpdateTodo command) {
+        Integer buffered = requestNext.getBufferedForEntitiesWithoutDemand().get(command.listId);
+        if (buffered != null && buffered >= 100) {
+          command.replyTo.tell(Response.REJECTED);
+        } else {
+          TodoList.Command requestMsg;
+          if (command.completed) requestMsg = new TodoList.CompleteTask(command.item);
+          else requestMsg = new TodoList.AddTask(command.item);
+          getContext()
+              .ask(
+                  Done.class,
+                  requestNext.askNextTo(),
+                  Duration.ofSeconds(5),
+                  askReplyTo ->
+                      new ShardingProducerController.MessageWithConfirmation<>(
+                          command.listId, requestMsg, askReplyTo),
+                  (done, exc) -> {
+                    if (exc == null) return new Confirmed(command.replyTo);
+                    else return new TimedOut(command.replyTo);
+                  });
+        }
+        return this;
+      }
+
+      private Behavior<Command> onConfirmed(Confirmed confirmed) {
+        confirmed.originalReplyTo.tell(Response.ACCEPTED);
+        return this;
+      }
+
+      private Behavior<Command> onTimedOut(TimedOut timedOut) {
+        timedOut.originalReplyTo.tell(Response.MAYBE_ACCEPTED);
+        return this;
+      }
+    }
+  }
+  // #producer
+
+  static void illustrateInit() {
+    Behaviors.setup(
+        context -> {
+          // #init
+          final DB db = null; // DB implementation
```

Comment:
"pretend we got it from somewhere" or something clarifying we don't actually mean it should be null

Reply:
fixed in https://github.com/akka/akka/pull/28155/commits/dba92f361636805efe4dd8f9b40982685dface95